torch/csrc/autograd/engine.cpp

```diff
@@ -158,6 +158,12 @@ int NodeTask::getReentrantDepth() const {
   return graph_task->reentrant_depth_;
 }
 
+bool graph_task_completed(const std::shared_ptr<GraphTask>& graph_task) {
+  std::unique_lock<std::mutex> lock(graph_task->mutex_);
```

Comment:
I don't really understand why you're taking out the lock now, when you didn't previously, because it seems like you didn't change the body of this function and it's still accessing atomics. What's going on? Do you still need the atomics now?

Reply:
You are correct, I don't need the lock here. I think I added it for extra safety but on closer inspection we don't really need the lock.